edx_rest_framework_extensions/jwt_decoder.py

```diff
@@ -69,34 +54,44 @@ def jwt_decode_handler(token):
         MissingRequiredClaimError: Either the exp or iat claims is missing from the JWT payload.
         InvalidTokenError: Decoding fails.
     """
+    jwt_issuer = get_first_jwt_issuer()
+    key_set = _get_signing_jwk_key_set(jwt_issuer)
```

Comment:
Nit: Proposal: move this and try/except to a new method like `_verify_jwt_and_signature(token, jwt_issuer)`.  The method name would make it more clear what `verify_compact()` is doing.